src/general/permutations/naive.rs

```diff
@@ -96,6 +96,12 @@ mod tests {
         }
     }
 
+    #[test]
+    fn empty_array() {
+        let empty: std::vec::Vec<u8> = vec![];
+        assert_eq!(permute_unique(&empty), vec![vec![]]);
```

Comment:
What do you think about explicitly testing `permute` against an empty input?

```suggestion
        assert_eq!(permute(&empty), vec![vec![]]);
        assert_eq!(permute_unique(&empty), vec![vec![]]);
```